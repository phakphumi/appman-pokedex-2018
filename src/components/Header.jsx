import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  justify-content: center;

  width: 100%;
  height: 100px;

  line-height: 100px;
  font-size: 40px;
  font-weight: bold;
`;

const Header = () => (<Container data-testid="Header-container">My Pokedex</Container>);

export default Header;