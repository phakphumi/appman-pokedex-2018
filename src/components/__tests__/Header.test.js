import React from 'react';
import { render } from '@testing-library/react';

import Header from '../Header';

import '@testing-library/jest-dom';

describe('<Header />', () => {
  it('should contain text content "My Pokedex"', () => {
    const { container, queryByTestId } = render(<Header />);

    expect(queryByTestId('Header-container')).toHaveTextContent('My Pokedex');

    expect(container).toMatchSnapshot()
  })  
});
