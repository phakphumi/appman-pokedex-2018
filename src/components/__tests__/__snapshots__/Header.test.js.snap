// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should contain text content "My Pokedex" 1`] = `
<div>
  <div
    class="sc-bdfBwQ bAYWQc"
    data-testid="Header-container"
  >
    My Pokedex
  </div>
</div>
`;
